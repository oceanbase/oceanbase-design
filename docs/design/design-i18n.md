---
title: 国际化
order: 4
group: 基础组件
---

`@oceanbase/design` 默认语言是中文，如果要支持其他语言，需要手动开启国际化。

## 开启国际化

可通过 ConfigProvider 全局配置开启国际化，`ConfigProvider` 使用 `React` 的 `context` 特性，只需在应用外围包裹一次即可全局生效。

```jsx | pure
import { ConfigProvider } from '@oceanbase/design';
import zhCN from '@oceanbase/design/es/locale/zh-CN';

return (
  <ConfigProvider locale={zhCN}>
    {...}
  </ConfigProvider>
);
```

## 使用示例

<!-- prettier-ignore -->
<code src="../../packages/design/src/locale/demo/basic.tsx"></code>

## 目前支持语言

目前支持以下语言：

| 语言     | 文件名 |
| -------- | ------ |
| 简体中文 | zh-CN  |
| 繁体中文 | zh-TW  |
| 英文     | en-US  |

如当前国际化语言有问题或者需要支持其他语言，欢迎反馈给我们。
