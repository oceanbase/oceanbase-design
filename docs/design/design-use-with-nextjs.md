---
title: 在 Next.js 中使用
order: 3
group: 基础组件
---

- 为了和 `@oceanbase/design` 搭配使用，需要安装 `file-loader` 依赖，并修改 Next.js 配置：

```bash
npm install file-loader --save-dev
```

```ts
const nextConfig = {
  // to transpile some special packages to es5
  transpilePackages: ['@oceanbase', 'query-string', '@ant-design', '@ant-design/cssinjs'],
  webpack: config => {
    // to handle @oceanbase/design built-in font files
    config.module.rules.push({
      test: /\.(woff|woff2|ttf?g)$/i,
      use: ['file-loader'],
    });
    return config;
  },
};

export default nextConfig;
```

- 完整配置和可运行的示例，可以参考我们提供的 [Next.js + @oceanbase/design 项目模板](https://stackblitz.com/edit/nextjs-oceanbase-design)。

<iframe src="https://stackblitz.com/edit/nextjs-oceanbase-design?embed=1&file=app%2Fpage.tsx"
     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
     title="@oceanbase/ui reproduction template"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>
