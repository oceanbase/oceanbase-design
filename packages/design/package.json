{
  "name": "@oceanbase/design",
  "version": "0.4.2",
  "description": "The Design System of OceanBase",
  "keywords": [
    "oceanbase",
    "design",
    "ui",
    "react",
    "component"
  ],
  "homepage": "https://github.com/oceanbase/oceanbase-design/packages/design",
  "repository": {
    "type": "git",
    "url": "git@github.com:oceanbase/oceanbase-design.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "main": "lib/index.js",
  "unpkg": "dist/design.min.js",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "files": [
    "es",
    "lib",
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "build": "father build && cp src/style/reset.css dist/"
  },
  "dependencies": {
    "@ant-design/cssinjs": "^1.22.0",
    "@oceanbase/aliyun-theme": "^0.1.6",
    "@oceanbase/icons": "workspace:^",
    "@oceanbase/util": "workspace:^",
    "ahooks": "^2.10.14",
    "antd": "^5.22.1",
    "classnames": "^2.5.1",
    "lodash": "^4.17.21",
    "lottie-web": "^5.12.2",
    "prop-types": "^15.8.1",
    "rc-util": "^5.43.0",
    "react-sticky-mouse-tooltip": "^0.0.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "antd-token-previewer": "^2.0.8",
    "rc-select": "^14.16.3",
    "rc-slider": "^11.1.7",
    "rc-table": "^7.48.1",
    "rc-tree-select": "^5.24.4"
  },
  "peerDependencies": {
    "react": ">=16.9.0",
    "react-dom": ">=16.9.0"
  }
}
