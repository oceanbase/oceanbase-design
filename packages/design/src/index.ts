import './global.css';

export * from 'antd';
export { version } from '../package.json';

export { default as Alert } from './alert';
export type { AlertProps } from './alert';

export { default as Button } from './button';
export type { ButtonProps } from './button';

export { default as Space } from './space';
export type { SpaceProps, SpaceSize } from './space';

export { Col, Row } from './grid';
export type { ColProps, RowProps, ColSize } from './grid';

export { default as Typography } from './typography';
export type { TypographyProps } from './typography';

export { default as Card } from './card';
export type { CardProps } from './card';

export { default as ConfigProvider } from './config-provider';
export type { ConfigProviderProps, ConfigConsumerProps, ThemeConfig } from './config-provider';

export { default as Descriptions } from './descriptions';
export type {
  DescriptionsProps,
  DescriptionsItemType,
  DescriptionsItemProps,
} from './descriptions';

export { default as Empty } from './empty';
export type { EmptyProps } from './empty';

export { default as Form } from './form';
export type { FormProps, FormItemProps } from './form';

export { default as Modal } from './modal';
export type { ModalProps, ModalFuncProps, ModalProgressProps } from './modal';

export { default as Drawer } from './drawer';
export type { DrawerProps } from './drawer';

export { message, notification, token } from './static-function';

export { default as List } from './list';
export type { ListProps } from './list';

export { default as Table } from './table';
export type { TableProps } from './table';

export { default as Tabs } from './tabs';
export type { TabsProps } from './tabs';

export { default as Tag } from './tag';
export type { TagProps } from './tag';

export { default as Input } from './input';
export type { InputProps } from './input';

export { default as InputNumber } from './input-number';
export type { InputNumberProps } from './input-number';

export { default as Radio } from './radio';
export type { RadioProps } from './radio';

export { default as Checkbox } from './checkbox';
export type { CheckboxProps } from './checkbox';

export { default as Switch } from './switch';
export type { SwitchProps } from './switch';

export { default as Select } from './select';
export type { SelectProps } from './select';

export { default as TreeSelect } from './tree-select';
export type { TreeSelectProps } from './tree-select';

export { default as Slider } from './slider';
export type { SliderSingleProps, SliderRangeProps } from './slider';

export { default as Tooltip } from './tooltip';
export type { TooltipProps } from './tooltip';

export { default as Segmented } from './segmented';
export type { SegmentedProps } from './segmented';

export { default as Breadcrumb } from './breadcrumb';
export type { BreadcrumbProps, BreadcrumbItemProps } from './breadcrumb';

export { default as Result } from './result';
export type { ResultProps, ResultType, ResultStatusType } from './result';

export { default as Spin } from './spin';
export type { SpinProps } from './spin';

export { default as Badge } from './badge';
export type { BadgeProps } from './badge';

export { default as Lottie } from './lottie';
export type { LottieProps, LottieRef } from './lottie';

export { default as theme } from './theme';

import theme from './theme';

const { useToken } = theme;
export { useToken };

export type { PresetStatusColorType } from 'antd/es/_util/colors';
export { genLargeStyle } from './_util/genStyle';
