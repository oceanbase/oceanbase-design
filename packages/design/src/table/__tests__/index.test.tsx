import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider, Table } from '@oceanbase/design';
import type { TableProps } from '@oceanbase/design';

const columns = [
  {
    title: 'Name',
    dataIndex: 'name',
  },
  {
    title: 'Age',
    dataIndex: 'age',
  },
  {
    title: 'Address',
    dataIndex: 'address',
  },
];

const dataSource = [];
for (let i = 1; i < 100; i++) {
  dataSource.push({
    key: i,
    name: `John ${i}`,
    age: 32,
    address: `${i} Road, Hangzhou, Zhejiang Province`,
  });
}

export const TableTest: React.FC<TableProps<any>> = props => (
  <Table dataSource={dataSource} columns={columns} {...props} />
);

describe('Table', () => {
  it('render', () => {
    const { container, asFragment } = render(<TableTest />);
    expect(container.querySelector('.ant-pagination')).toBeTruthy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('render empty', () => {
    const { container } = render(<TableTest dataSource={[]} />);
    expect(container.querySelector('.ant-table-empty-wrapper')).toBeTruthy();
  });

  it('hideOnSinglePage should be false by default', () => {
    const { container, asFragment } = render(<TableTest dataSource={dataSource.slice(0, 10)} />);
    expect(container.querySelector('.ant-pagination')).toBeTruthy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('hideOnSinglePage could be changed', () => {
    const { container, asFragment } = render(
      <TableTest
        dataSource={dataSource.slice(0, 10)}
        pagination={{
          showSizeChanger: false,
          // hideOnSinglePage work only when showSizeChanger is false
          hideOnSinglePage: true,
        }}
      />
    );
    expect(container.querySelector('.ant-pagination')).toBeFalsy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('row selection should work', () => {
    const { container, asFragment } = render(
      <TableTest
        rowSelection={{
          selectedRowKeys: ['1', '2'],
        }}
      />
    );
    // selection column
    expect(container.querySelector('.ant-table-selection-column')).toBeTruthy();
    // batch operation bar
    expect(container.querySelector('.ant-table-batch-operation-bar')).toBeTruthy();
  });

  it('default pagination should work', () => {
    const { container, asFragment } = render(<TableTest />);
    // pagination.showTotal
    expect(container.querySelector('.ant-pagination-total-text').textContent).toBe('99 in Total');
    // pagination.showSizeChanger
    expect(container.querySelector('.ant-pagination-options')).toBeTruthy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('default empty should work', () => {
    const { container, asFragment } = render(<TableTest dataSource={[]} />);
    expect(container.querySelector('.ant-empty-image')).toBeTruthy();
    expect(container.querySelector('.ant-empty-description')).toBeTruthy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('ConfigProvider pagination should work', () => {
    const { container, asFragment } = render(
      <ConfigProvider
        pagination={{
          showTotal: undefined,
          showSizeChanger: false,
        }}
      >
        <TableTest />
      </ConfigProvider>
    );
    // pagination.showTotal
    expect(container.querySelector('.ant-pagination-total-text').textContent).toBe('');
    // pagination.showSizeChanger
    expect(container.querySelector('.ant-pagination-options')).toBeFalsy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
