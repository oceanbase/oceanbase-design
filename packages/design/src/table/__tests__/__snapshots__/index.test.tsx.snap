// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table > hideOnSinglePage could be changed 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
          >
            <table
              style="table-layout: auto;"
            >
              <colgroup />
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Age
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Address
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 1
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    1 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 2
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    2 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="3"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 3
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    3 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="4"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 4
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    4 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="5"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 5
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    5 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="6"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 6
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    6 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="7"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 7
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    7 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="8"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 8
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    8 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="9"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 9
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    9 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="10"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 10
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    10 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table > hideOnSinglePage should be false by default 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
          >
            <table
              style="table-layout: auto;"
            >
              <colgroup />
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Age
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Address
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 1
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    1 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 2
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    2 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="3"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 3
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    3 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="4"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 4
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    4 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="5"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 5
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    5 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="6"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 6
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    6 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="7"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 7
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    7 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="8"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 8
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    8 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="9"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 9
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    9 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="10"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 10
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    10 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination ant-table-pagination-right"
      >
        <li
          class="ant-pagination-total-text"
        >
          <span />
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-prev ant-pagination-disabled"
          title="Previous Page"
        >
          <button
            class="ant-pagination-item-link"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="left"
              class="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-next ant-pagination-disabled"
          title="Next Page"
        >
          <button
            class="ant-pagination-item-link"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="right"
              class="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Table > render 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
          >
            <table
              style="table-layout: auto;"
            >
              <colgroup />
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Age
                  </th>
                  <th
                    class="ant-table-cell"
                    scope="col"
                  >
                    Address
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 1
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    1 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 2
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    2 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="3"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 3
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    3 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="4"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 4
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    4 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="5"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 5
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    5 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="6"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 6
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    6 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="7"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 7
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    7 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="8"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 8
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    8 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="9"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 9
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    9 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="10"
                >
                  <td
                    class="ant-table-cell"
                  >
                    John 10
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    32
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    10 Road, Hangzhou, Zhejiang Province
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination ant-table-pagination-right"
      >
        <li
          class="ant-pagination-total-text"
        >
          <span />
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-prev ant-pagination-disabled"
          title="Previous Page"
        >
          <button
            class="ant-pagination-item-link"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="left"
              class="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-2"
          tabindex="0"
          title="2"
        >
          <a
            rel="nofollow"
          >
            2
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-3"
          tabindex="0"
          title="3"
        >
          <a
            rel="nofollow"
          >
            3
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-4"
          tabindex="0"
          title="4"
        >
          <a
            rel="nofollow"
          >
            4
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
          tabindex="0"
          title="5"
        >
          <a
            rel="nofollow"
          >
            5
          </a>
        </li>
        <li
          class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
          tabindex="0"
          title="Next 5 Pages"
        >
          <a
            class="ant-pagination-item-link"
          >
            <div
              class="ant-pagination-item-container"
            >
              <span
                aria-label="double-right"
                class="anticon anticon-double-right ant-pagination-item-link-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="double-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                  />
                </svg>
              </span>
              <span
                class="ant-pagination-item-ellipsis"
              >
                •••
              </span>
            </div>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-10"
          tabindex="0"
          title="10"
        >
          <a
            rel="nofollow"
          >
            10
          </a>
        </li>
        <li
          aria-disabled="false"
          class="ant-pagination-next"
          tabindex="0"
          title="Next Page"
        >
          <button
            class="ant-pagination-item-link"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="right"
              class="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          class="ant-pagination-options"
        >
          <div
            aria-label="Page Size"
            class="ant-select ant-select-outlined ant-pagination-options-size-changer ant-select-single ant-select-show-arrow ant-select-show-search"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-label="Page Size"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  role="combobox"
                  type="search"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="10 / page"
              >
                10 / page
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
